cpp/tests/testnninputs.cpp

```diff
@@ -1118,3 +1118,191 @@ void Tests::runBasicSymmetryTests() {
   }
 
 }
+
+void Tests::runBoardSymmetricTests() {
+  cout << "Running board Symmetric tests" << endl;
+  ostringstream out;
+//============================================================================
+  auto printMarkedSymDupArea = [&out](const Board& board, const bool * isSymDupLoc){
+    for(int y = 0; y<board.y_size; y++) {
+      for(int x = 0; x<board.x_size; x++) {
+        Loc loc = Location::getLoc(x,y,board.x_size);
+        if (isSymDupLoc[loc]) {
+          out << 'x';
+        } else {
+          out << '.';
+        }
+      }
+      out << endl;
+    }
+    out << endl;
+  };
+  const char* name = "Symmetric Board Test";
```

Comment:
This is a neat set of test cases! Can you add one that is 4-way rotationally symmetric that does not have any reflection symmetry?

Reply:
Two additional tests are added.